core/src/text/fontContext.cpp

```diff
@@ -14,98 +14,95 @@
+            LOG("Adding default system font");
 
-    std::string fallback = "";
-    int importance = 0;
+            for (int i = 0, size = BASE_SIZE; i < MAX_STEPS; i++, size += STEP_SIZE) {
+                m_font[i] = m_alfons.addFont("default", alfons::InputSource(systemFont), size);
+            }
+        } else {
+            size_t dataSize;
+            char* data = reinterpret_cast<char*>(bytesFromFile(DEFAULT, dataSize));
 
-    while (importance < 100) {
-        fallback = systemFontFallbackPath(importance++, 400);
-        if (fallback.empty()) { break; }
+            if (data) {
+                LOG("Loading default font file %s", DEFAULT);
 
-        if (fallback.find("UI-") != std::string::npos) {
-            continue;
-        }
-        fontPath = ANDROID_FONT_PATH;
-        fontPath += fallback;
-        LOGD("FALLBACK %s", fontPath.c_str());
+                for (int i = 0, size = BASE_SIZE; i < MAX_STEPS; i++, size += STEP_SIZE) {
+                    m_font[i] = m_alfons.addFont("default", alfons::InputSource(data, dataSize), size);
+                }
 
-        int size = BASE_SIZE;
-        for (int i = 0; i < MAX_STEPS; i++, size += STEP_SIZE) {
-            m_font[i]->addFace(m_alfons.addFontFace(alfons::InputSource(fontPath), size));
-        }
-    }
-#elif defined(PLATFORM_IOS)
-
-    int size = BASE_SIZE;
-    auto loadFonts = [&](const char* path) {
-        size_t dataSize;
-        char* data = reinterpret_cast<char*>(bytesFromFile(path, dataSize));
-        if (data) {
-            for (int i = 0; i < MAX_STEPS; i++, size += STEP_SIZE) {
-                m_font[i] = m_alfons.addFont("default", alfons::InputSource(data, dataSize), size);
+                free(data);
+            } else {
+                LOGW("Default font %s not found", DEFAULT);
             }
-            free(data);
         }
-    };
-    auto addFaces = [&](const char* path) {
-        size_t dataSize;
-        char* data = reinterpret_cast<char*>(bytesFromFile(path, dataSize));
-        if (data) {
-            for (int i = 0; i < MAX_STEPS; i++, size += STEP_SIZE) {
-                    m_font[i]->addFace(m_alfons.addFontFace(alfons::InputSource(data, dataSize), size));
+    }
+
+    // Treat fallback fonts
+    {
+        std::string fallback = "";
+        int importance = 0;
+        fallback = systemFontFallbackPath(importance++, DEFAULT_BOLDNESS);
+
+        if (fallback.empty()) {
+            LOGW("No system font fallback, loading bundled fonts");
+
+            auto addFaces = [&](const char* path) {
+                size_t dataSize;
+                char* data = reinterpret_cast<char*>(bytesFromFile(path, dataSize));
+
+                if (data) {
+                    LOG("Adding bundled font at path %s", path);
+
+                    for (int i = 0, size = BASE_SIZE; i < MAX_STEPS; i++, size += STEP_SIZE) {
+                        m_font[i]->addFace(m_alfons.addFontFace(alfons::InputSource(data, dataSize), size));
```

Comment:
So, with this change if default font could not be added, this line will crash. I hit this issue when getting tizen branch upto date with master.